sync3/handler/connstate.go

```diff
@@ -93,6 +93,9 @@ func (s *ConnState) load(ctx context.Context, req *sync3.Request) error {
 	for _, metadata := range joinedRooms {
 		metadata.RemoveHero(s.userID)
 		urd := s.userCache.LoadRoomData(metadata.RoomID)
+		// TODO: is this necessary---will the joinNID already be in the UserCache?
+		urd.JoinNID = joinNIDs[metadata.RoomID]
+
```

Comment:
Not sure on this one.

Reply:
No, it won't be. `userCache.LoadRoomData` _lazily_ loads data, which is seeded from `(h *SyncLiveHandler) userCache(userID string)`. As you don't seed the join nids in that function, it will be unset. Seeding happens by calling `userCache.On....` callbacks from data from the database in order to reuse codepaths.

By setting this here, this will not take effect in the user cache (`LoadRoomData` returns copies of metadata precisely to avoid callers polluting the cache). To get this to stick around, you need to add a new function to `UserCache` which will assign the join nids and then call that in the handler to seed the cache correctly. It's a bit bleh, since the user cache doesn't know the list of joined rooms at that point in time, so you'll need to hit `s.globalCache.LoadJoinedRooms(ctx, s.userID)` twice for new conns (once to seed the cache, once for the conn).


EDIT: I wrote all of this, and then read the code in `OnRegistered`, which does exactly this (`LoadJoinedRooms` then sets join nids). So ignore the above.